src/app/workflow/workflow-detail/workflow-detail.component.ts

```diff
@@ -391,6 +349,114 @@ export class WorkflowDetailComponent implements OnInit, OnDestroy {
       });
   }
 
+
+  openCopy(content, jobId: string) {
+    this.getJob(content, jobId);
+  }
+
+  getJob(content, jobId: string ) {
+    this.jobService.getJob(jobId).subscribe(job => {
+        this.jobToCopy = job;
+        this.jobService.getPlugin(job.wippExecutable).subscribe(plugin => {
+          this.selectedSchema = this.pluginList.find(x => x.id === plugin.id);
+          this.jobModel['taskName'] = this.jobToCopy.name + '-copy';
+          this.jobModel['inputs'] = {};
+          const requests = [];
+          for (const input of Object.keys(this.jobToCopy.parameters)) {
+            if ((this.selectedSchema.properties.inputs.properties[input]['widget'] === 'search' || this.selectedSchema.properties.inputs.properties[input]['widget']['id'] === 'search') && this.jobToCopy.parameters[input].indexOf('{') === -1) {
```

Comment:
Could you add comments for the if/else if/else? For example for the if "Input to copy is an existing WIPP object"
And also, could you break the if into separate lines? That would make it more readable